pgx-tests/src/tests/array_tests.rs

```diff
@@ -99,6 +100,52 @@ fn return_zero_length_vec() -> Vec<i32> {
     Vec::new()
 }
 
+#[pg_extern]
+fn get_arr_nelems(arr: Array<i32>) -> libc::c_int {
+    // SAFETY: Eh it's fine, it's just a len check.
+    unsafe { RawArray::from_array(arr).unwrap().len() }
+}
+
+/// # Safety
+/// Don't call with an actually-null index, or you might see an invalid value.
+#[pg_extern]
+#[deny(unsafe_op_in_unsafe_fn)]
+unsafe fn get_arr_data_ptr_nth_elem(arr: Array<i32>, elem: i32) -> Option<i32> {
+    // SAFETY: this is Known to be an Array from ArrayType,
+    // and the index has to be a valid one.
+    unsafe {
+        let raw = RawArray::from_array(arr).unwrap().data::<i32>();
+        let slice = &(*raw.as_ptr());
+        slice.get(elem as usize).copied()
+    }
+}
+
+#[pg_extern]
+fn display_get_arr_nullbitmap(arr: Array<i32>) -> String {
+    let raw = unsafe { RawArray::from_array(arr) }.unwrap();
+
+    if let Some(slice) = raw.nulls() {
+        // SAFETY: If the test has gotten this far, the ptr is good for 0+ bytes.
+        let slice = unsafe { &*slice.as_ptr() };
```

Comment:
`&*slice.as_ptr()`  👀

Reply:
This pattern is called a "reborrow".
It brings us from `NonNull<[T]>` to `&[T]`.

Reply:
Ah, more reading material. Thanks 😄